src/Grids/convenience_constructors.jl

```diff
@@ -0,0 +1,366 @@
+    Nq::Integer,
+    h_elem::Integer,
+    device::ClimaComms.AbstractDevice = ClimaComms.device(),
+    context::ClimaComms.AbstractCommsContext = ClimaComms.context(device),
+    quad::Quadratures.QuadratureStyle = Quadratures.GLL{Nq}(),
+    h_mesh = Meshes.EquiangularCubedSphere(
+        Domains.SphereDomain{FT}(radius),
+        h_elem,
+    ),
+) where {FT}
+    check_device_context(context, device)
+    h_topology = Topologies.Topology2D(context, h_mesh)
+    return Grids.SpectralElementGrid2D(h_topology, quad)
+end
+
+"""
+    ColumnGrid(
+        ::Type{<:AbstractFloat}; # defaults to Float64
+        z_elem::Integer,
+        z_min::Real,
+        z_max::Real,
+        device::ClimaComms.AbstractDevice = ClimaComms.device(),
+        context::ClimaComms.AbstractCommsContext = ClimaComms.context(device),
+        stretch::Meshes.StretchingRule = Meshes.Uniform(),
+    )
+
+A convenience constructor, which builds a
+`FiniteDifferenceGrid` given.
+"""
+ColumnGrid(; kwargs...) = ColumnGrid(Float64; kwargs...)
+function ColumnGrid(
+    ::Type{FT};
+    z_elem::Integer,
+    z_min::Real,
+    z_max::Real,
+    device::ClimaComms.AbstractDevice = ClimaComms.device(),
+    context::ClimaComms.AbstractCommsContext = ClimaComms.context(device),
+    stretch::Meshes.StretchingRule = Meshes.Uniform(),
+) where {FT}
+    check_device_context(context, device)
+    z_boundary_names = (:bottom, :top)
+    z_domain = Domains.IntervalDomain(
+        Geometry.ZPoint{FT}(z_min),
+        Geometry.ZPoint{FT}(z_max);
+        boundary_names = z_boundary_names,
+    )
+    z_mesh = Meshes.IntervalMesh(z_domain, stretch; nelems = z_elem)
+    z_topology = Topologies.IntervalTopology(context, z_mesh)
+    return FiniteDifferenceGrid(z_topology)
+end
+
+"""
+    Box3DGrid(
+        ::Type{<:AbstractFloat}; # defaults to Float64
+        z_elem::Integer,
+        x_min::Real,
+        x_max::Real,
+        y_min::Real,
+        y_max::Real,
+        z_min::Real,
+        z_max::Real,
+        x1periodic::Bool,
+        x2periodic::Bool,
+        Nq::Integer,
+        n1::Integer,
+        n2::Integer,
+        device::ClimaComms.AbstractDevice = ClimaComms.device(),
+        context::ClimaComms.AbstractCommsContext = ClimaComms.context(device),
+        stretch::Meshes.StretchingRule = Meshes.Uniform(),
+        hypsography::HypsographyAdaption = Flat(),
+        global_geometry::Geometry.AbstractGlobalGeometry = Geometry.CartesianGlobalGeometry(),
+        quad::Quadratures.QuadratureStyle = Quadratures.GLL{Nq}(),
+    )
+
+A convenience constructor, which builds a
+`ExtrudedFiniteDifferenceGrid` with a
+`FiniteDifferenceGrid` vertical grid and a
+`SpectralElementGrid2D` horizontal grid.
+"""
+Box3DGrid(; kwargs...) = Box3DGrid(Float64; kwargs...)
+function Box3DGrid(
+    ::Type{FT};
+    z_elem::Integer,
+    x_min::Real,
+    x_max::Real,
+    y_min::Real,
+    y_max::Real,
+    z_min::Real,
+    z_max::Real,
+    x1periodic::Bool,
+    x2periodic::Bool,
```

Comment:
```suggestion
    periodic_x::Bool,
    periodic_y::Bool,
```
seems a little clearer to me

Reply:
Sorry it's been some time. Done!